### PR TITLE
Simplify the DosFilter for #1256

### LINKED ISSUE #1256: DoSFilter leaks USER_AUTH entries
`DoSFilter` automatically removes `RateTracker` entries that belong to type `USER_IP` or `USER_SESSION`, but not the other types.

`DoSFilter` also needs a code review because for example exposes `RateTracker` as a return type of a public method, but `RateTracker` is package private and so it's not possible to override the method.

### PR DESCRIPTION
**Pull request by gregw**
Use only IP tracking for the DosFilter to fix #1256

### CODE REVIEW

File: jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java [hunk 1]
```diff
@@ -523,28 +501,10 @@ protected void onRequestTimeout(HttpServletRequest request, HttpServletResponse
...
-            return RateType.AUTH;
-        if (tracker != null)
-            return tracker.getType();
-        return RateType.UNKNOWN;
-    }
-
-    /**
-     * @return the maximum priority that we can assign to a request
-     */
+    @Deprecated
```
Comment (lorban) [inline jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java:508]: That's missing a javadoc tag explaining why it's deprecated.

File: jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java [hunk 2]
```diff
@@ -775,13 +703,7 @@ protected void stopScheduler()
...
     }
 
-    /**
-     * Returns the user id, used to track this connection.
-     * This SHOULD be overridden by subclasses.
-     *
-     * @param request the current request
-     * @return a unique user id, if logged in; otherwise null.
-     */
+    @Deprecated
```
Comment (lorban) [inline jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java:715]: That's missing a javadoc tag explaining why it's deprecated.

File: jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java [hunk 3]
```diff
@@ -1503,13 +1368,13 @@ public Action onRequestOverLimit(HttpServletRequest request, OverLimit overlimit
             switch (action)
             {
                 case REJECT:
-                    LOG.warn("DOS ALERT: Request rejected ip={}, overlimit={}, session={}, user={}", request.getRemoteAddr(), overlimit, request.getRequestedSessionId(), request.getUserPrincipal());
+                    LOG.warn("DOS ALERT: Request rejected ip={}, overlimit={}, user={}", request.getRemoteAddr(), overlimit, request.getUserPrincipal());
```
Comment (lorban) [inline jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java]: Nit: denial of service is usually stylized as `DoS`, while `DOS` refers to everyone's favorite operating system of the 80's.

File: jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java [hunk 4]
```diff
@@ -750,7 +678,7 @@ private byte[] prefixToBytes(int prefix, int length)
             return result;
 
         // Sets the _prefix_ most significant bits to 1
-        result[index] = (byte)~((1 << (8 - prefix)) - 1);
+        result[index] = (byte)-(1 << (8 - prefix));
```
Comment (lorban) [inline jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java:685]: Any reason why you changed the bit manipulation? It looks correct, but it's always hard to review without tests.

**Review by lorban — CHANGES_REQUESTED** [review]
Just a few nits.

File: jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java (hunk 4 above)
Comment (gregw) [reply jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java:685]: To keep intellij happy

File: documentation/jetty-documentation/src/main/asciidoc/old_docs/extras/dos-filter.adoc [hunk 5]
```diff
@@ -32,9 +32,7 @@ The filter works on the assumption that the attacker might be written in simple
 [[dos-filter-using]]
 ==== Using the DoS Filter
 
-Jetty places throttled requests in a priority queue, giving priority first to authenticated users and users with an HttpSession, then to connections identified by their IP addresses.
-Connections with no way to identify them have lowest priority.
-To uniquely identify authenticated users, you should implement the The extractUserId(ServletRequest request) function.
+Jetty places throttled requests in a queue, and proceed only when there is capacity available.
```
Comment (sbordet) [inline documentation/jetty-documentation/src/main/asciidoc/old_docs/extras/dos-filter.adoc:35]: This is the old documentation for Jetty 9.4.x, so leave it unchanged.

**Review by lorban — APPROVED** [review]

**Review by sbordet — CHANGES_REQUESTED** [review]

**Review by sbordet — APPROVED** [review]

**Pull request merged**